cfgov/unprocessed/js/organisms/FilterableListControls.js

```diff
@@ -18,157 +17,254 @@ var Notification = require( '../molecules/Notification' );
 
     if ( validatedFields.invalid.length > 0 ) {
-      _showError();
+      event.preventDefault();
+      var msg = '';
+
+      validatedFields.invalid.forEach( function( validation ) {
+        msg += validation.label + validation.msg + '</br>';
+      } );
+      _setNotification( _notification.ERROR, msg ||
+                        ERROR_MESSAGES.DEFAULT );
+    }
+  }
+
+  /**
+   * Validate the fields of our form.
+   * @param {Field} field A form field.
+   * @param {String} selector Selector used to retreive the dom element.
+   * @param {Boolean} isInGroup Flag used determine if field is in group.
+   * @returns {String} The label of the field.
+   */
+  function _getLabelText( field, selector, isInGroup ) {
+    var labelText = '';
+    var labelDom;
+
+    if ( isInGroup && !selector ) {
+      labelDom = getClosestElement( field, 'fieldset' );
+      if ( labelDom ) labelDom = labelDom.querySelector( 'legend' );
     } else {
-      _showSuccess();
+      selector = selector ||
+                 'label[for="' + field.getAttribute( 'id' ) + '"]';
+      labelDom = _form.querySelector( selector );
     }
+
+    if ( labelDom ) labelText = labelDom.textContent.trim();
+
+    return labelText;
   }
 
   /**
-   * Show error notification.
-  */
-  function _showError() {
-    _notification.setTypeAndContent( _notification.ERROR, 'Error!' );
-    _notification.show();
+   * Set the notification type, msg, and visibility.
+   * @param {String} type The type of notification to display.
+   * @param {String} msg The message to display in the notification.
+   * @param {String} methodName The method to use to control visibility
+                                of the notification.
+   */
+  function _setNotification( type, msg, methodName ) {
+    methodName = methodName || 'show';
+    _notification.setTypeAndContent( type, msg );
+    _notification[methodName]();
   }
 
   /**
-   * Show success notification.
-  */
-  function _showSuccess() {
-    _notification.setTypeAndContent( _notification.SUCCESS, 'Success!' );
-    _notification.show();
+   * Determines if you should validate a field.
+   * @param {Field} field A form field.
+   * @param {String} type The type of field.
+   * @param {Boolean} isInGroup A boolean that determines if field in a group.
+   * @returns {Boolean} Value indicating whether to validate a field.
+   */
+  function shouldValidateField( field, type, isInGroup ) {
+    var isDisabled = field.getAttribute( 'disabled' ) !== null;
+    var isIgnoreType = _defaults.ignoreFieldTypes.indexOf( type ) > -1;
+    var shouldValidate = isDisabled === false && isIgnoreType === false;
+
+    if ( shouldValidate && ( isInGroup || false ) ) {
+      var name = field.getAttribute( 'data-group' ) ||
+                 field.getAttribute( 'name' );
+      var isRequired = field.getAttribute( 'data-required' ) !== null;
+      if ( _fieldGroups.indexOf( name ) > -1 || isRequired === false ) {
```

Comment:
fixed in https://github.com/cfpb/cfgov-refresh/commit/f217937d4eec3537f28765b2f956d29b1609df50